src/components/implementation/no_interface/cbboot/lock.c

```diff
@@ -183,7 +183,7 @@ static inline void __lock_help(struct meta_lock *l, int smp)
 	do {
 		prev_val.v         = l->lock_addr->v;
 		owner              = prev_val.c.owner;
-		assert(owner); /* from lock_help_owner, this must be true */
+		if (owner) continue;
```

Comment:
I guess I've honestly never thought about how `continue` and `do ... while` interact.